### ISSUE TITLE
perf(inference): batch prefill instead of token-by-token prompt ingestion

### ISSUE THREAD

**Issue by sankpal-shreyas** [Shreyas582/WraithRun#62]
﻿## Problem

Both inference paths in onnx_vitis.rs feed prompt tokens **one at a time** instead of as a single batch:

**session.run path** (~line 2655):
`
ust
for (prompt_index, prompt_token) in context_ids.iter().copied().enumerate() {
    let step_input_ids = vec![prompt_token]; // SINGLE TOKEN per forward pass
    let model_inputs = build_model_inputs(&layout, &step_input_ids, ...)?;
    let mut outputs = session.run(model_inputs)?;
}
`

**
un_prompt_shared_buffer IoBinding path** (~line 2486):
`
ust
for (prompt_index, prompt_token) in context_ids.iter().copied().enumerate() {
    let step_input_ids = vec![prompt_token]; // SINGLE TOKEN per forward pass
    rebind_step_inputs(&mut binding, layout, &step_input_ids, ...)?;
    let outputs = session.run_binding(&binding)?;
}
`

For an ~800 token prompt at ~35ms per ORT forward pass = **~28 seconds just for prompt ingestion** on a 1B q4 CPU model. This makes the CI E2E action time out on larger models.

## Expected Behavior

Pass all prompt tokens in a **single** forward pass (batch prefill):
`
ust
let model_inputs = build_model_inputs(&layout, &context_ids, context_ids.len(), false, &cache_state)?;
let mut outputs = session.run(model_inputs)?;
// Extract KV-cache from this single pass, then decode token-by-token
`

select_next_token already handles multi-position logits correctly (takes last ocab_size slice).

ebind_step_inputs already handles multi-token step_input_ids for position_ids and cache_position.

## Measured Impact

| Metric | Current (token-by-token) | Expected (batch) |
|---|---|---|
| Prefill time (1B q4, ~800 tokens, CPU) | ~28s | <0.5s |
| First token latency (1B q4, 1 step) | ~42s | <15s |
| CI E2E (7B hybrid, Vitis) | Timeout (>45min) | Should complete |

## Scope

- [ ] Implement batch prefill in session.run path (non-shared-buffer)
- [ ] Implement batch prefill in 
un_prompt_shared_buffer IoBinding path
- [ ] Verify KV-cache outputs are correctly extracted from batched prefill pass
- [ ] Benchmark before/after with q4, int8, and full-precision models
